### ISSUE TITLE
[MobileDatePicker] Broken ARIA reference 

### ISSUE THREAD

**Issue by 212817** [mui/mui-x#5632]
### Duplicates

- [X] I have searched the existing issues

### Latest version

- [X] I have tested the latest version

### Current behavior 😯

When scanning for accessibility issues using the app Wave. The MobileDatePicker generate the follow issue: 
"Broken ARIA reference "

 

### Expected behavior 🤔

The MobileDatePicker should not generate any accessibility issues.

### Steps to reproduce 🕹

Link to live example:
https://codesandbox.io/s/responsivedatepickers-demo-mui-x-forked-wuwiu7?file=/demo.tsx

Steps:
1. Install the Wave extension in a chrome browser.
2. Click on the MobileDatePicker  or DatePiker with props desktopModeMediaQuery='never' to open a selection window
3. Scan the page with a Date Picker on.

![MicrosoftTeams-image](https://user-images.githubusercontent.com/35614882/181229191-f99fa17f-3dd6-4504-af28-f441fcd8a005.png)

### Your environment 🌎

<details>
  <summary><code>npx @mui/envinfo</code></summary>

```
  Don't forget to mention which browser you used.
  Output from `npx @mui/envinfo` goes here.
```
</details>

**Comment by cherniavskii** [comment]
Thanks @212817 
There's missing `aria-labelledby` prop passed to `Dialog` here:
https://github.com/mui/mui-x/blob/bc263d50b42ed2a068024a6d773f0efc0ac8d808/packages/x-date-pickers/src/internals/components/PickersModalDialog.tsx#L69

that should point to the id of the title component in here:
https://github.com/mui/mui-x/blob/bc263d50b42ed2a068024a6d773f0efc0ac8d808/packages/x-date-pickers/src/internals/components/PickersToolbar.tsx#L111-L112
